src/lib/handlers/handle_ping_start.go

```diff
@@ -31,10 +33,18 @@ func HandlePingStart(config parser.YAMLBenchRoutesType, pingServiceState string)
 		}
 	}
 	doPing(config, urlStack, pingInterval)
+	for {
+		select {
+		case <-ctxGeneralPing.Done():
+			fmt.Println("General Stopper Called")
+			return
+		}
+	}
```

Comment:
You sure, this shouldnot not be in an infinite loop? Because I think we should be listenting to the context the whole time, and the for{} makes sure its listening. Whats your opinion. @Harkishen-Singh

Reply:
okay. will confirm after having a closer look at this.